packages/hawtio/src/plugins/camel/routes-service.ts

```diff
@@ -212,15 +212,12 @@ class RoutesService {
 
   async dumpRoutesStatsXML(routesNode: MBeanNode): Promise<string | null> {
     let xml = null
-    const routeNodeOperation = 'dumpRouteStatsAsXml'
     const routesFolderOperation = 'dumpRoutesStatsAsXml'
-
     const mbeanName = routesNode.getMetadata(contextNodeType)
-    const operationForMBean = mbeanName ? routesFolderOperation : routeNodeOperation
-    const mbeanToQuery = mbeanName ? mbeanName : (routesNode.objectName ?? '')
+    const mbeanToQuery = mbeanName ? mbeanName : (routesNode.parent?.getMetadata(contextNodeType) ?? '')
 
     try {
-      xml = await jolokiaService.execute(mbeanToQuery, operationForMBean, [true, true])
+      xml = await jolokiaService.execute(mbeanToQuery, routesFolderOperation, [true, true])
```

Comment:
Now this name `routesFolderOperation` doesn't tell the intent clearly. Better to rename it to avoid confusion for code readers.